### PR TITLE
Add unit test for DBImplSecondary

### PR DESCRIPTION
**Pull request by riversand963**
Summary: as title

Test Plan:
```
$make clean && make -j32 db_secondary_test
$./db_secondary_test
$make -j32 all check
```
All tests must pass.

### CODE REVIEW

File: db/db_secondary_test.cc [hunk 1]
```diff
@@ -0,0 +1,217 @@
...
+
+  auto db_secondary = static_cast<DBImplSecondary*>(db1);
+  ASSERT_NE(nullptr, db_secondary);
+  ReadOptions ropts;
+  ropts.verify_checksums = true;
+  std::string value;
+  ASSERT_NOK(db_secondary->Get(ropts, "foo", &value));
+  ASSERT_NOK(db_secondary->Get(ropts, "bar", &value));
+
+  ASSERT_OK(db_secondary->TryCatchUpWithPrimary());
```
Comment (siying) [inline db/db_secondary_test.cc]: I'm confused by this interface. TryCatchUpWithPrimary() is not a public function, so who is supposed to call it when users run a secondary?

Comment (riversand963) [reply db/db_secondary_test.cc]: I'm going to add a thread calling this function periodically to update version storage info. It's not public intentionally.

Comment (riversand963) [reply db/db_secondary_test.cc]: Update: this function is now public.

**Comment by riversand963** [comment]
Moved to #4899

**Pull request closed without merging**